### PR TITLE
#14422. Avoid crash with an invalid reference in SetSessionDescriptionObserver::OnFailure

### PR DESCRIPTION
**Pull request by alber2510**

### CODE REVIEW

File: src/rtcModule/webrtcAdapter.h [hunk 1]
```diff
@@ -146,9 +146,9 @@ class SdpSetCallbacks : public webrtc::SetSessionDescriptionObserver
     virtual void OnFailure(const std::string& error)
     {
         RTCM_DO_CALLBACK(
-             mPromise.reject(::promise::Error(error, kSetSdpDescriptionFailed, ERRTYPE_RTC));
+             mPromise.reject(::promise::Error("Unkown error", kSetSdpDescriptionFailed, ERRTYPE_RTC));
```
Comment (sergiohs84) [inline src/rtcModule/webrtcAdapter.h]:
```suggestion
             mPromise.reject(::promise::Error("Unknown error", kSetSdpDescriptionFailed, ERRTYPE_RTC));
```

File: src/rtcModule/webrtcAdapter.h [hunk 2]
```diff
@@ -146,9 +146,9 @@ class SdpSetCallbacks : public webrtc::SetSessionDescriptionObserver
...
              Release();
-        , this, error);
+        , this, "Unkown error");
```
Comment (sergiohs84) [inline src/rtcModule/webrtcAdapter.h]:
```suggestion
        , this, "Unknown error");
```

**Comment by sergiohs84** [comment]
WebRTC says the interface we are using is deprecated. Please, test with:
```
void OnFailure(RTCError error)
```
If it works, better to close this PR and create a new one. Otherwise, it's good as it is (just fix the typo).

**Comment by alber2510** [comment]
Use method `void OnFailure(RTCError error)` is better option, I close this PR and open a new one

**Pull request closed without merging**